crates/sdk/src/queries/vp/pos.rs

```diff
@@ -249,6 +254,44 @@ where
     read_all_validator_addresses(ctx.state, epoch)
 }
 
+/// Get liveness information for all consensus validators in the current epoch.
+fn liveness_info<D, H, V, T>(
+    ctx: RequestCtx<'_, D, H, V, T>,
+) -> namada_storage::Result<LivenessInfo>
+where
+    D: 'static + DB + for<'iter> DBIter<'iter> + Sync,
+    H: 'static + StorageHasher + Sync,
+{
+    let epoch = ctx.state.in_mem().last_epoch;
+    let consensus_validators =
+        read_consensus_validator_set_addresses(ctx.state, epoch)?;
+    let params = read_pos_params::<_, governance::Store<_>>(ctx.state)?;
+
+    let mut result = vec![];
```

Comment:
one more small thing, now that the size of this vec is same as `consensus_validators.len()` we can allocate it to its size